src/Mod/PartDesign/App/FeatureSketchBased.cpp

```diff
@@ -665,6 +708,154 @@ void ProfileBased::generatePrism(TopoDS_Shape& prism,
     }
 }
 
+void ProfileBased::generateTaperedPrism(const TopoDS_Shape& sketchshape,
+                                        const gp_Dir& direction,
+                                        const double Length,
+                                        const double Length2,
+                                        const double angle,
+                                        const double angle2,
+                                        std::list<TopoDS_Shape>& drafts)
+{
+    // inspect sketchshape
+    // that it is not null was already checked
+    TopoDS_Wire sourceWire;
+    if (sketchshape.ShapeType() == TopAbs_WIRE) {
+        ShapeFix_Wire aFix;
+        aFix.Load(TopoDS::Wire(sketchshape));
+        aFix.FixReorder();
+        aFix.FixConnected();
+        aFix.FixClosed();
+        sourceWire = aFix.Wire();
+    }
+    else if (sketchshape.ShapeType() == TopAbs_FACE) {
+        TopoDS_Wire outerWire = ShapeAnalysis::OuterWire(TopoDS::Face(sketchshape));
+        sourceWire = outerWire;
+    }
+    else if (sketchshape.ShapeType() == TopAbs_COMPOUND) {
+        TopoDS_Iterator it(sketchshape);
+        for (; it.More(); it.Next()) {
+            generateTaperedPrism(it.Value(), direction, Length, Length2, angle, angle2, drafts);
+        }
+    }
+    else {
+        throw Base::RuntimeError("Only a wire or a face is supported");
+    }
+
+    if (sourceWire.IsNull()) {
+        throw Base::RuntimeError("Offset for base shape could not made");
+    }
+
+    // calculate lateral offset and vectors for top and bottom shape of the loft
+    double lateralOffset = tan(Base::toRadians(angle)) * Length;
+    double lateralOffset2 = tan(Base::toRadians(angle2)) * Length2;
+    gp_Vec vecOffset = gp_Vec(direction) * Length;
+    gp_Vec vecOffset2 = gp_Vec(direction.Reversed()) * Length2;
+
+    // create a new shape which is the input shape with the offset defined by the TaperAngle
+    // at first inspect the from shape
+    int numEdges = 0;
+    TopExp_Explorer xp(sourceWire, TopAbs_EDGE);
+    while (xp.More()) {
+        numEdges++;
+        xp.Next();
+    }
+    // Pad / Pocket does not only support closed wires, thus only 1 edge is not valid

```

Comment:
does only

Reply:
Wait a second, what if I'm padding a circle? wouldn't that be just one edge but still closed? 1 edge should be valid, and isn't the shape already validated to be closed before this?

Reply:
> what if I'm padding a circle?

This is a special case because OCC has a bug in making offsets of circles.

However, this PR has to be remodeled to take the code from Part Extrude. -> I'll put it on hold until I found the time to write the helper class to achieve this.